### DISCUSSION TITLE
How can I replace the code behind c# class of the view component?

### DISCUSSION THREAD

**Discussion by ertugrulbakidal** [abpframework/abp#10338] [Q&A]
I am able to replace AccountProfilePasswordManagementGroup's Html and js by creating the same folder structure in my project but I'd like to replace AccountProfilePasswordManagementGroupViewComponent c# class itself and don't know how to replace it. From Abp documentation "If you need, you can also replace the code behind c# class of the component just using the dependency injection system.".

**Comment by iyilm4z** [comment]
Hi @ertugrulbaki27, i know you already came here from the same doc but the part titled [Overriding a Page](https://docs.abp.io/en/abp/latest/UI/AspNetCore/Customization-User-Interface#overriding-a-page) at the doc shows how to do what you exactly need. However you probably didn't understand the sentence below there.

> If you need, you can also replace the code behind c# class of the component just using the dependency injection system.

This means they already showed how to replace page's pagemodel class. So why they need to write almost the same how-to-replace doc for viewcomponent class as well? They probably need to replace that sentence with a clearer one for preventing this kinda lack of understandings.

Also this thread should be under issues tab not under discusssions tab because there is nothing to discuss on your statement:)

**Reply by ertugrulbakidal** [reply]
Hey @iyilm4z,
Thanks for your answer. I've created an [issue ](https://github.com/abpframework/abp/issues/10343). If you want, you can follow up.
